src/NzbDrone.Api/Music/ArtistBulkImportModule.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Nancy;
+using NzbDrone.Api.REST;
+using NzbDrone.Api.Extensions;
+using NzbDrone.Core.MediaCover;
+using NzbDrone.Core.MetadataSource;
+using NzbDrone.Core.Parser;
+using System.Linq;
+using System;
+using Marr.Data;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Datastore;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.MediaFiles.EpisodeImport;
+using NzbDrone.Core.RootFolders;
+using NzbDrone.Common.Cache;
+using NzbDrone.Core.Music;
+
+namespace NzbDrone.Api.Music
+{
+
+    public class UnmappedComparer : IComparer<UnmappedFolder>
+    {
+        public int Compare(UnmappedFolder a, UnmappedFolder b)
+        {
+            return a.Name.CompareTo(b.Name);
+        }
+    }
+
+    public class MusicBulkImportModule : NzbDroneRestModule<ArtistResource>
+    {
+        private readonly ISearchForNewSeries _searchProxy;
+        private readonly IRootFolderService _rootFolderService;
+        private readonly IMakeImportDecision _importDecisionMaker;
+        private readonly IDiskScanService _diskScanService;
+		private readonly ICached<Core.Music.Artist> _mappedArtists;
+        private readonly IArtistService _artistService;
+
+        public MusicBulkImportModule(ISearchForNewSeries searchProxy, IRootFolderService rootFolderService, IMakeImportDecision importDecisionMaker,
```

Comment:
May be worth splitting this. Github won't show the whole line in the diff, but I'm not sure what happens with the normal code browser. Lines 37 and 48 look like they aren't aligned with the other lines if we want to fix that.